### PR TITLE
Add Voice Arena read endpoints

### PR DESCRIPTION
**Pull request by seribaymadina**
Adds the read-only API for the Voice Arena (layer 4 of the build plan).

## Endpoints
- `GET /v1/arena/battle` returns one battle to vote on
- `GET /v1/arena/battle/{id}` returns a specific battle
- `GET /v1/arena/leaderboard?metric=&domain=` returns the latest computed board

- Battles are served blind: the response omits provider/model identities so voting stays unbiased. The A/B to model mapping stays server-side.
- The leaderboard returns the rows sharing the most recent `computed_at` for the requested metric and domain, sorted by Elo. It is empty until the snapshot job (a later layer) has run.
- Battle selection is a placeholder (uniform random). Adaptive pairing replaces it in a later layer; the endpoint contract does not change.
- Reads the pool directly with raw SQL like the other routers, so this lands independently of the DB access layer PR.

## Release Notes

* **New Features**
  * Added Arena endpoints enabling blind battle voting and leaderboard queries
  * Blind battles conceal model identities to ensure unbiased voting
  * Leaderboard displays model ratings and performance statistics, filterable by metric and domain
  * Rate limiting applied (60 requests per minute per endpoint)

<h3>Greptile Summary</h3>

This PR adds three read-only Voice Arena endpoints (`GET /v1/arena/battle`, `GET /v1/arena/battle/{id}`, and `GET /v1/arena/leaderboard`) along with their Pydantic schemas and a thorough test suite. Battles are served blind (provider/model identities stripped), and the leaderboard returns the most recent computed snapshot for a given metric and domain.

- **Blind battle selection** uses a placeholder `ORDER BY random()` (acknowledged in comments) with all model identity columns withheld from the response schema.
- **Leaderboard SQL** uses a CTE to isolate the single most recent `(computed_at, methodology_version)` pair per metric/domain before joining back for the full row set, correctly preventing mixed-version boards.
- **Test coverage** is comprehensive: blindness invariant, 404/422 edge cases, domain/metric scoping, staleness exclusion, and methodology-version tiebreaking are all verified with a real in-process Postgres instance.

<h3>Confidence Score: 5/5</h3>

Safe to merge — all three endpoints use parameterized queries, the blind invariant is enforced at the SQL column-selection layer, and the leaderboard CTE correctly isolates single-version boards.

The core logic is correct: battles are served blind, leaderboard filtering by metric/domain is applied both in the CTE and outer WHERE, and all DB access is parameterized. Two minor observations (lexicographic version tiebreaker and unconstrained metric string) do not affect current correctness.

runner/src/coval_bench/api/routers/arena.py — the methodology_version tiebreaker and metric param type are worth a second look before the leaderboard snapshot job ships.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| runner/src/coval_bench/api/routers/arena.py | New router implementing three blind read endpoints; SQL and parameterization are correct, minor lexicographic tiebreaker and unconstrained metric-param concerns noted. |
| runner/src/coval_bench/api/schemas.py | Adds BattleOut, LeaderboardEntryOut, and ArenaLeaderboardResponse schemas; intentional float wins/losses/ties for fractional scoring as clarified in prior review thread. |
| runner/tests/api/test_arena.py | Comprehensive test coverage: blindness, 404/422 edge cases, domain/metric filtering, latest-board selection, and methodology-version tiebreaker are all exercised. |
| runner/src/coval_bench/api/app.py | Minimal change: imports the arena router and mounts it at /v1; no issues. |

</details>

<h3>Sequence Diagram</h3>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
sequenceDiagram
    participant Client
    participant ArenaRouter
    participant PgPool as Postgres Pool
    participant PostHog

    Client->>ArenaRouter: GET /v1/arena/battle
    ArenaRouter->>PgPool: SELECT id, prompt_text, domain, audio_a_url, audio_b_url FROM arena.battles ORDER BY random() LIMIT 1
    PgPool-->>ArenaRouter: row (or None)
    alt No rows
        ArenaRouter-->>Client: 404 no battles available
    else Row found
        ArenaRouter->>PostHog: capture arena_battle_served
        ArenaRouter-->>Client: 200 BattleOut (blind)
    end

    Client->>ArenaRouter: "GET /v1/arena/battle/{id}"
    ArenaRouter->>PgPool: "SELECT ... FROM arena.battles WHERE id = %(id)s"
    PgPool-->>ArenaRouter: row (or None)
    alt Not found
        ArenaRouter-->>Client: 404
    else Found
        ArenaRouter-->>Client: 200 BattleOut (blind)
    end

    Client->>ArenaRouter: "GET /v1/arena/leaderboard?metric=&domain="
    ArenaRouter->>PgPool: CTE picks latest (computed_at, methodology_version), JOIN snapshots WHERE metric/domain match, ORDER BY rating_elo DESC
    PgPool-->>ArenaRouter: board rows
    ArenaRouter->>PostHog: capture arena_leaderboard_queried
    ArenaRouter-->>Client: 200 ArenaLeaderboardResponse
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
sequenceDiagram
    participant Client
    participant ArenaRouter
    participant PgPool as Postgres Pool
    participant PostHog

    Client->>ArenaRouter: GET /v1/arena/battle
    ArenaRouter->>PgPool: SELECT id, prompt_text, domain, audio_a_url, audio_b_url FROM arena.battles ORDER BY random() LIMIT 1
    PgPool-->>ArenaRouter: row (or None)
    alt No rows
        ArenaRouter-->>Client: 404 no battles available
    else Row found
        ArenaRouter->>PostHog: capture arena_battle_served
        ArenaRouter-->>Client: 200 BattleOut (blind)
    end

    Client->>ArenaRouter: "GET /v1/arena/battle/{id}"
    ArenaRouter->>PgPool: "SELECT ... FROM arena.battles WHERE id = %(id)s"
    PgPool-->>ArenaRouter: row (or None)
    alt Not found
        ArenaRouter-->>Client: 404
    else Found
        ArenaRouter-->>Client: 200 BattleOut (blind)
    end

    Client->>ArenaRouter: "GET /v1/arena/leaderboard?metric=&domain="
    ArenaRouter->>PgPool: CTE picks latest (computed_at, methodology_version), JOIN snapshots WHERE metric/domain match, ORDER BY rating_elo DESC
    PgPool-->>ArenaRouter: board rows
    ArenaRouter->>PostHog: capture arena_leaderboard_queried
    ArenaRouter-->>Client: 200 ArenaLeaderboardResponse
```

</a>

<sub>Reviews (2): Last reviewed commit: ["Select a single leaderboard board so met..."](https://github.com/coval-ai/benchmarks/commit/13454e3875709da7c5c1b40a0f808c792151a967) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=38702413)</sub>

### CODE REVIEW

File: runner/src/coval_bench/api/schemas.py [hunk 1]
```diff
@@ -165,3 +166,40 @@ class LeaderboardResponse(BaseModel):
...
+    provider: str
+    model: str
+    rating_elo: float
+    rating_bt: float
+    ci_low: float | None
+    ci_high: float | None
+    ci_half_width: float | None
+    votes_total: int
+    wins: float
+    losses: float
```
[the remark this answers is not in the document]
Comment (seribaymadina) [reply runner/src/coval_bench/api/schemas.py:193]: intentional

File: runner/src/coval_bench/api/routers/arena.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+from coval_bench.api.ratelimit import limiter
+from coval_bench.api.schemas import ArenaLeaderboardResponse, BattleOut, LeaderboardEntryOut
+
+logger = structlog.get_logger("coval_bench.api")
+
+router = APIRouter(tags=["arena"])
+
+
+# Columns exposed by the (blind) battle endpoints — provider/model are withheld.
+_BATTLE_COLS = "id, prompt_text, domain, audio_a_url, audio_b_url"
```
Comment (coval-cale) [inline runner/src/coval_bench/api/routers/arena.py:42]: I think the audio urls are built as https://*/{provider}-{model}.wav or something like that so it would be pretty easy to figure out that info. I'm not super concerned but definitely something to be aware of.

**Review by coval-cale — APPROVED** [review]
YEWWW!

File: runner/src/coval_bench/api/routers/arena.py (hunk 2 above)
Comment (seribaymadina) [reply runner/src/coval_bench/api/routers/arena.py:42]: super helpful feedback, will be careful with this in audio generation stage of arena! for now letting it pass since I will have to ground more decisions at that stage
